### PR TITLE
feat: added UpdateBar ui component

### PR DESCRIPTION
**Pull request by natto1784**
This patch adds the UI component "UpdateBar"

### CODE REVIEW

File: packages/ui/src/components/molecules/UpdateBar.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
+import React, { FC } from 'react';
+import styled from 'styled-components';
+import { Typography, Flex, Image } from '../atoms';
+
+interface UpdateBarProps {
+  text: string;
+  button?: string;
```
Comment (irshadCypherock) [inline packages/ui/src/components/molecules/UpdateBar.tsx]:
```suggestion
  buttonText?: string;
```

File: packages/ui/src/components/molecules/UpdateBar.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  handler?: React.MouseEventHandler<HTMLButtonElement>;
```
Comment (irshadCypherock) [inline packages/ui/src/components/molecules/UpdateBar.tsx]:
```suggestion
  onButtonClick?: React.MouseEventHandler<HTMLButtonElement>;
```

File: packages/ui/src/components/molecules/UpdateBar.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  progress?: number;
+  icon?: string;
+  state?: 'normal' | 'progress' | 'success' | 'error';
```
Comment (irshadCypherock) [inline packages/ui/src/components/molecules/UpdateBar.tsx]: Define and export a type for state

File: packages/ui/src/components/molecules/UpdateBar.tsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+const UpdateBarButton = styled.button`
+  background: none;
+  outline: none;
+  border: none;
+  &:hover {
+    filter: brightness(150%);
+    cursor: pointer;
+  }
+`;
```
Comment (irshadCypherock) [inline packages/ui/src/components/molecules/UpdateBar.tsx]: Create a new variant in the existing Button component.

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

File: packages/ui/src/components/molecules/UpdateBar.tsx (hunk 4 above)
Comment (natto1784) [reply packages/ui/src/components/molecules/UpdateBar.tsx]: sure but ill have to override certain properties like padding and transition

File: packages/ui/src/components/atoms/Button.tsx [hunk 5]
```diff
@@ -20,7 +20,7 @@ interface ButtonProps
     UtilsProps,
     SpacingProps,
     React.ButtonHTMLAttributes<HTMLButtonElement> {
-  variant?: 'primary' | 'secondary' | 'dashed' | 'warning' | 'none';
+  variant?: 'primary' | 'secondary' | 'dashed' | 'warning' | 'update' | 'none';
```
Comment (irshadCypherock) [inline packages/ui/src/components/atoms/Button.tsx]: A more appropriate name for this variant would be `text`

File: packages/ui/src/components/molecules/UpdateBar.tsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
+import React, { FC } from 'react';
+import styled from 'styled-components';
+import { Typography, Flex, Image, Button } from '../atoms';
+
+export type updateState = 'normal' | 'progress' | 'success' | 'error';
```
Comment (irshadCypherock) [inline packages/ui/src/components/molecules/UpdateBar.tsx]: Type should be in `PascalCase`, check [CODING GUIDELINES](https://github.com/Cypherock/cypherock-cysync/blob/develop/docs/CODING_GUIDELINES.md#naming-convention)

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

**Review by irshadCypherock — APPROVED** [review]

**Pull request merged**